prisma/migrations/20250402214855_enable_rls/migration.sql

```diff
@@ -0,0 +1,274 @@
+CREATE POLICY "Service role access all WalletExports" ON "WalletExports" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all WorkKeyShares" ON "WorkKeyShares" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all RecoveryKeyShares" ON "RecoveryKeyShares" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all Challenges" ON "Challenges" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all AnonChallenges" ON "AnonChallenges" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all DevicesAndLocations" ON "DevicesAndLocations" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all Sessions" ON "Sessions" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all ApplicationSessions" ON "ApplicationSessions" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all LoginAttempts" ON "LoginAttempts" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all Organizations" ON "Organizations" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all Teams" ON "Teams" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+CREATE POLICY "Service role access all Memberships" ON "Memberships" 
+  USING (current_setting('request.jwt.claims', true)::json->>'role' = 'service_role');
+
+-- User-specific policies
+-- UserProfiles
+CREATE POLICY "Users can view their own profile" ON "UserProfiles"
+  FOR SELECT USING (auth.uid() = "supId");
+CREATE POLICY "Users can update their own profile" ON "UserProfiles"
+  FOR UPDATE USING (auth.uid() = "supId");
+
+-- Wallets
+CREATE POLICY "Users can view their own wallets" ON "Wallets"
+  FOR SELECT USING (auth.uid() = "userId");
+CREATE POLICY "Users can update their own wallets" ON "Wallets"
+  FOR UPDATE USING (auth.uid() = "userId");
+CREATE POLICY "Users can insert their own wallets" ON "Wallets"
+  FOR INSERT WITH CHECK (auth.uid() = "userId");
+CREATE POLICY "Users can delete their own wallets" ON "Wallets"
+  FOR DELETE USING (auth.uid() = "userId");
+
+-- WalletActivations
+CREATE POLICY "Users can view their own wallet activations" ON "WalletActivations"
+  FOR SELECT USING (auth.uid() = "userId");
+CREATE POLICY "Users can insert their own wallet activations" ON "WalletActivations"
```

Comment:
@matteyu Should updates in certain tables like this one be disallowed altogether?

In any case, we can improve these policies on a new PR on top of this one. I think the sooner we start testing using a backend with this enabled, the better, in case any of them is too restrictive.

Reply:
Regarding deletes, not sure who should be allowed. If we could just have a cronjob that removes rows older than some value on certain non-critic tables (WalletActivations, WalletRecoveries, WalletExports... that are just informational), then we should probably disallow it for users.